doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,224 @@
+
+### Custom Test Definition
+
+* A custom test is any sort of test logic required, written in any programming language that satifies the scorecard test requirements for output format and execution.
+
+### Custom Test Components
+
+This proposal outlines the end-user facing components that make up a custom test including:
+
+#### Test Image
+
+A custom test is based on a container image, that image is created by
+the end-user using whatever tools they want to use.
+
+The test image must produce log output that conforms to the scorecard Test Result output.
+
+The test image will be executed as a Pod by the scorecard with a restart policy of `never`.
+
+When executing a scorecard test, scorecard will create a ConfigMap holding the scorecard configuration manifests such as the Custom Resource or Bundle being tested.  This ConfigMap can then be mounted into the custom test image and used within the test execution.  
+
+#### Test Output
+
+Custom tests are expected to produce log output that conforms to the
+scorecard v1alpha2 Test Result output.  This allows scorecard to parse
+the custom test log output and aggregate the results like any other
+scorecard test.
+
+The scorecard expected output would be in JSON, and would match the following definition:  https://github.com/operator-framework/operator-sdk/blob/master/pkg/apis/scorecard/v1alpha2/types.go#L36
+
+The scorecard v1alpha2 Test Result output contains information such as
+pass or fail status and suggestions/errors produced by the test that
+the end user would want to see in the aggregated scorecard results.
+
+
+### Custom Test Development Model
+
+The developer of a custom test would have a workflow similar to:
+ 
+ * create a Pod manifest to run their custom test image, copying a sample Pod manifest from the scorecard examples in the SDK repository
+ * write a custom test binary in the language of their choice, sample custom test within the SDK repository would be available as a starting point
+ * test the custom test binary locally (out-of-cluster) to make sure the output is produced that matches v1alpha2 ScorecardTestResult format
+ * when the test binary works as expected, the developer would build and push the test image to a container registry accessible to the scorecard test environment
+ * the scorecard configuration would be updated to reference the custom test Pod manifest location
+ * the scorecard would discover the custom test based on the scorecard configuration and also the custom test labels applied by the end-user within the custom test Pod manifest
+
+### Custom Test Packaging and Configuration
+
+In this design proposal, custom tests are Pod manifests in YAML that run a custom test image.
+
+Note that is would also be possible to add a resource limit to the scorecard configuration as shown in the example below.  Those resource limits would be applied to custom tests that are executed allowing for some control as to how much resources are available for the test execution.
```

Comment:
typo: "Note that is would" should be "Note that it would"

Reply:
fixed in latest commit.